internal/storage/fs/store/store.go

```diff
@@ -251,6 +253,12 @@ func newObjectStore(ctx context.Context, cfg *config.Config, logger *zap.Logger)
 		return nil, err
 	}
 
+	if prefix != "" {
+		// ensure prefix is a valid prefix
+		prefix = strings.TrimPrefix(prefix, "/") + "/"
```

Comment:
This could be a breaking change for someone. Let say they set prefix to `production` and have files like `production.flipt.yml` and `production.features.yml` in their bucket. This change probably will break their setup. I believe if you want  a folder structure you have to explicitly add `/` at the end in object storage world.

Reply:
thats a good point. we could instead specify that you need to add the `/` to your prefix in the config if you want to use 'folders' and remove this logic